### PR TITLE
T-11 Add functionality to create new asset in portfolio

### PR DESCRIPTION
**Pull request by DashaDrahun**
Functionality to create concrete types of assets in portfolio

### CODE REVIEW

**Comment by stas-litvinenka** [comment]
[T-11 Add functionality to create new asset in portfolio](https://trello.com/c/wE72HjeK/11-t-11-add-functionality-to-create-new-asset-in-portfolio)

File: PortfolioKeeper/BusinessLogic/Services/AssetService.cs [hunk 1]
```diff
@@ -18,19 +23,85 @@ public class AssetService : IAssetService
...
+        }
+
+        public List<AssetPropertyTypeDto> GetAssetPropertyTypes(byte childrenAssetTypeId)
+        {
+            var allProperties = new List<AssetPropertyType>();
+            var childrenProperties =
+                assetPropertyTypeRepository.Get(e => e.AssetTypeId == childrenAssetTypeId).ToList();
+            allProperties.AddRange(childrenProperties);
+            byte? parentTypeId = GetParentTypeIdByChildrenTypeId(childrenAssetTypeId);
+            if (parentTypeId == null) throw new ArgumentNullException(nameof(parentTypeId));
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Services/AssetService.cs]: not valid type of exception here. parentTypeId  is not argument

File: PortfolioKeeper/BusinessLogic/Services/AssetService.cs [hunk 2]
```diff
@@ -18,19 +23,85 @@ public class AssetService : IAssetService
...
+            allProperties.AddRange(parentProperties);
+            return customMapper.MapAssetPropertyType(allProperties);
+        }
+
+        public void Create(List<AssetPropertyValueViewModel> properties, string userId, string assetName,
+            byte assetTypeId)
+        {
+            var portfolio = portfolioRepository.Get(x => x.AppUserId == userId).FirstOrDefault();
+            if (portfolio == null) throw new ArgumentNullException(nameof(portfolio));
+            if (properties == null) throw new ArgumentNullException(nameof(properties));
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Services/AssetService.cs]: please move it above

File: PortfolioKeeper/BusinessLogic/Services/AssetService.cs [hunk 3]
```diff
@@ -18,19 +23,85 @@ public class AssetService : IAssetService
...
+            if (userId == null) throw new ArgumentNullException(nameof(userId));
```
Comment (stas-litvinenka) [inline PortfolioKeeper/BusinessLogic/Services/AssetService.cs]: same. also please change check to string.isnullorwhitespace

File: PortfolioKeeper/Data/Context/FluentApiConfiguration/AssetTypeMap.cs [hunk 4]
```diff
@@ -18,10 +18,10 @@ public AssetTypeMap()
                 .HasMaxLength(50)
                 .IsRequired();
 
-            Property(e => e.DisplayName)
-                .HasColumnType("NVARCHAR")
-                .HasMaxLength(50)
-                .IsRequired(); 
+            //Property(e => e.DisplayName)
```
Comment (stas-litvinenka) [inline PortfolioKeeper/Data/Context/FluentApiConfiguration/AssetTypeMap.cs]: rename if needed

**Comment by DashaDrahun** [comment]
resolved issues from Stas review comments

File: PortfolioKeeper/BusinessLogic/ServiceMappers/AssetServiceMapper.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
+﻿using System;
+using System.Collections.Generic;
+using PortfolioKeeper.BusinessLogic.Dto;
+using PortfolioKeeper.Data.Entities;
+using PortfolioKeeper.Models.Asset;
+
+namespace PortfolioKeeper.BusinessLogic.ServiceMappers
+{
+    public class AssetServiceMapper:IAssetServiceMapper
```
Comment (AntonBessarabov) [inline PortfolioKeeper/BusinessLogic/ServiceMappers/AssetServiceMapper.cs]: Add spaces, please

File: PortfolioKeeper/Controllers/AssetController.cs [hunk 6]
```diff
@@ -1,45 +1,124 @@
...
-        public AssetController(IAssetService assetService, IMapper mapper)
+        public AssetController(IAssetService assetService, AddingAssetHelper assetHelper, AssetMapper assetMapper, IMapper mapper)
         {
             this.assetService = assetService ?? throw new ArgumentNullException(nameof(assetService));
+            this.helper = assetHelper ?? throw new ArgumentNullException(nameof(assetHelper));
+            this.mapperHelper = assetMapper ?? throw new ArgumentNullException(nameof(assetMapper));
             this.mapper = mapper ?? throw new ArgumentNullException(nameof(mapper));
         }
 
+        private void AddConcreteAsset<T>(T asset, byte childAssetTypeId) where T : BaseAssetViewModel
```
Comment (AntonBessarabov) [inline PortfolioKeeper/Controllers/AssetController.cs]: Move  `where T : BaseAssetViewModel` to a new line, please

**Pull request merged**